File: updated_code/code/python/1128-number-of-equivalent-domino-pairs.py
from typing import List

class Solution:
    def numEquivDominoPairs(self, dominoes: List[List[int]]) -> int:
        """
        This function calculates the number of pairs of dominoes in a given list that are equivalent.

        Args:
            dominoes (List[List[int]]): A list of dominoes, where each domino is a list of two integers.

        Returns:
            int: The number of pairs of dominoes that are equivalent.
        """
        count = {}
        for domino in dominoes:
            # sort the domino to make it unique
            domino.sort()
            # count the frequency of each domino
            count[tuple(domino)] = count.get(tuple(domino), 0) + 1
        pairs = 0
        for v in count.values():
            # calculate the number of pairs for each unique domino
            pairs += v * (v - 1) // 2
        return pairs

    from typing import List

    class Solution:
        def numEquivDominoPairs(self, dominoes: List[List[int]]) -> int:

    class Solution:
        """
        This class contains a method to calculate the number of pairs of dominoes in a given list that are equivalent.
        """

        def numEquivDominoPairs(self, dominoes: List[List[int]]) -> int:
            """
            This function calculates the number of pairs of dominoes in a given list that are equivalent.

            Args:
                dominoes (List[List[int]]): A list of dominoes, where each domino is a list of two integers.

            Returns:
                int: The number of pairs of dominoes that are equivalent.
            """
    
    def num_equiv_domino_pairs(self, dominoes):
    
    from typing import List
    
    pairs += v * (v - 1) // 2
    
    return pairs
    # This is the last line of the code. Adding a newline now.
    
    return pairs
    # This is the last line of the code. Adding a newline.
    
    """
    This module contains a class to calculate the number of pairs of dominoes in a given list that are equivalent.
    """
    class Solution:
        """
        This class contains a method to calculate the number of pairs of dominoes in a given list that are equivalent.
        """
    
    def num_equiv_domino_pairs(self, dominoes: List[List[int]]) -> int:
        """
        This function calculates the number of pairs of dominoes in a given list that are equivalent.

        Args:
            dominoes (List[List[int]]): A list of dominoes, where each domino is a list of two integers.

        Returns:
            int: The number of pairs of dominoes that are equivalent.
        """

File: updated_code/code/java/1128-number-of-equivalent-domino-pairs.java
package com.example;

import java.util.HashMap;
import java.util.Map;

/**
 * This class is used to count the number of equivalent domino pairs.
 */
public class Solution {
    public int numEquivDominoPairs(int[][] dominoes) {
        // Use a HashMap to store the count of each domino
        Map<String, Integer> map = new HashMap<>();
        for (int[] domino : dominoes) {
            // Create a key for the domino by sorting the values and joining them with a comma
            String key = String.format("%d,%d", Math.min(domino[0], domino[1]), Math.max(domino[0], domino[1]));
            // Increment the count for the key
            map.put(key, map.getOrDefault(key, 0) + 1);
        }

        int res = 0;
        for (int count : map.values()) {
            // Calculate the number of pairs for each count
            res += count * (count - 1) / 2;
        }
        return res;
    }
}